DuckDuckGo/RemoteMessaging/TabBarRemoteMessageView.swift

```diff
@@ -0,0 +1,118 @@
+    @State private var wasCloseButtonHovered: Bool = false
+
+    let model: TabBarRemoteMessage
+
+    let onClose: () -> Void
+    let onTap: (URL) -> Void
+    let onHover: () -> Void
+    let onHoverEnd: () -> Void
+    let onAppear: () -> Void
+
+    var body: some View {
+        HStack(spacing: 0) {
+            HStack {
+                Text(model.buttonTitle)
+                    .font(.system(size: 13))
+                    .fixedSize(horizontal: true, vertical: false)
+                    .foregroundColor(.white)
+            }
+            .padding([.leading, .top, .bottom], 8)
+            .padding(.trailing, 6)
+            .cornerRadius(8)
+            .background(wasViewHovered
+                        ? Color("PrimaryButtonHover")
+                        : Color("PrimaryButtonRest"))
+            .onTapGesture { onTap(model.surveyURL) }
+            .onHover { hovering in
+                wasViewHovered = hovering
+
+                if hovering {
+                    onHover()
+                } else {
+                    onHoverEnd()
+                }
+            }
+
+            Divider()
+                .background(Color.white.opacity(0.3))
+                .frame(width: 1)
+                .padding([.top, .bottom], 3)
+
+            HStack {
+                Image(.close)
+                    .resizable()
+                    .scaledToFit()
+                    .foregroundColor(.white)
+                    .frame(width: 16, height: 16)
+            }
+            .padding([.top, .bottom])
+            .padding([.leading, .trailing], 4)
+            .background(wasCloseButtonHovered
+                        ? Color("PrimaryButtonHover")
+                        : Color("PrimaryButtonRest"))
+            .cornerRadius(8)
+            .onTapGesture {
+                onClose()
+            }
+            .onHover { hovering in
+                wasCloseButtonHovered = hovering
+            }
+            .frame(maxWidth: .infinity)
+        }
+        .background(wasCloseButtonHovered || wasViewHovered
+                    ? Color("PrimaryButtonHover")
+                    : Color("PrimaryButtonRest"))
+        .frame(height: 24)
+        .cornerRadius(8)
+        .onAppear(perform: { onAppear() })
+    }
+}
+
+struct TabBarRemoteMessagePopoverContent: View {
+    let model: TabBarRemoteMessage
+
+    var body: some View {
+        HStack(alignment: .center, spacing: 12) {
+            Image(.daxResponse)
+                .resizable()
+                .scaledToFit()
+                .frame(width: 72, height: 72)
+
+            VStack(alignment: .leading, spacing: 8) {
+                Text(model.popupTitle)
+                    .font(.system(size: 13, weight: .bold))
+                    .padding(.top, 9)
+
+                Text(model.popupSubtitle)
+                    .font(.system(size: 13, weight: .medium))
+                    .padding(.bottom, 9)
+            }
+        }
+        .frame(width: 360)
```

Comment:
Now, we only fix the width of the popover; if the text is larger, the popover should increase its height.